_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,102 @@
+logic), I ask. If the review already has many comments, I might do this
+outside the review itself.<br />
+Also, if I see code that appears to be copied, where I feel improvements can be
+made, I ask if this is something that has been considered/if this is something
+for us to consider in a follow-up pull request.
+- **I re-read a review before I submit it.** I don't always do this, but I
+know it helps. I've seen the author ask about comments I wrote, not (fully)
+understanding what I meant. Though communication is hard, I feel this has
+been happening less over time.
+- **I feel responsible for the code I review,** meaning I use "we", not
+"you". "We could consider adding a comment explaining why we need to ignore X",
+not "you".
+- **I review my own code.** Especially in larger pull requests, I look through
+the code before assigning a reviewer, asking open questions to the reviewer,
+making notes and/or making some changes. Seeing the "bigger" picture often helps
+me catch issues/improvements to be made early, hopefully making the code review
+more enjoyable in the process.
+- **I don't use words like "easy", "simply", or similar.** The fact something
+would be any of those is always an assumption to whoever is reading it, and
+serves no purpose.
+
+---
+
+Keep in mind that this is not an exhaustive list, but rather a bunch of the
+principles I try to keep in mind while reviewing, and there's still so much to
+learn. If you have any thoughts on this post, like guidelines that you're using
+that I've missed here, I'd love to hear about them!
+
+Bonus: I'm a big fan of a tool we built at work to get a random reviewer for
+your code. It's called `lotto`, and picks a random reviewer outside of your
+team. It's great.
+
+## On Non-Code Reviews
+
+Oh, and most of this also applies reviewing other things, like blog posts and
+documentation. For those in particular, I prefer printing the documents (so it
+is easy to make both corrections and remarks). That also giving me the
+opportunity to find another place to work, away from my computer. Natural light
+helps!
+
+If possible, I always discuss it in person afterwards.
+
+---
+
+If you're interested in code review or proofreading of a document, I'm more
+than happy to help out. Just send me a direct message [on Twitter](https://twitter.com/{{ site.twitter_username }}).[^2]
+
+<sub>Thanks _so much_ to [Kristaps](https://twitter.com/fassko),
+[Donny](https://twitter.com/DonnyWals), [Fernando](https://twitter.com/fmo_91),
+[João](https://twitter.com/NSMyself), [Barbie](https://twitter.com/barbieinbeta)
+and [Mustapha](https://twitter.com/ben_muta) for proofreading!</sub>
+
+[^1]: I suppose "better" is always subjective here, but I can take the learnings from these reviews and the insights of the current code base to help make certain decisions.
+[^2]: I find it frustrating that asking for these things myself is arguably one of my weaker points. 🤷‍♂️
```

Comment:
Good post! I think too many people focus on PRs and code reviews as ways to point out mistakes to people, and too little on learning from each other, respecting each other's work and effort.

One thing that I would love to read your opinion on is what to do with "nits". Sometimes there are small changes that should be made to a PR to make it perfect / good. Like small changes to phrasing in documentation comments. I usually just add those as comments and approve the PR anyway. Curious to see what you think

Reply:
Thanks!

On nits:

First of all, "to make it perfect" is impossible. 😄

I think nits are an important part of a code review, but require (thorough) documentation on their expectations. I believe they should never block a pull request from being merged, and should be distinguishable from other comments (I know I brought up the documentation, so this is somewhat ironic, but we use 💅 as an unwritten rule to distinguish nits from other comments).

I wouldn't want a reviewer not to point out these small things, as in my experience in the worst case they make something just a bit better, and in the best case make you think about it a bit more, realizing a bigger improvement can be made.

On approving: I also tend to just "comment" on a pull request if there are many open questions/comments, rather than "requesting changes". That said, it would also be something we could document better so that its meaning is better understood by others.